File: 0367-valid-perfect-square/0367-valid-perfect-square.php
class Solution {

    /**
     * @param Integer $num
     * @return Boolean
     */
    function isPerfectSquare($num) {
        if($num < 2){
            return true;
        }
        
        $left = 1;
        $right = $num / 2;
        
        while($left <= $right){
            $middle = intval(($left + $right) / 2);
            $sqrt = $middle * $middle;
                        
            if($sqrt == $num){
                return true;
            }
            
            if($sqrt < $num){
                $left = $middle + 1; 
            } else {
                $right = $middle - 1;
            }
        }
        
        return false;
    }
}

File: 0367-valid-perfect-square/0367-valid-perfect-square.go
func isPerfectSquare(num int) bool {
    if num < 2 {
        return true
    }
    
    l, r := 0, num / 2
    
    for l <= r {
        mid := (r + l) / 2
        sqrt := mid * mid
        
        if sqrt == num {
            return true
        }
        
        if sqrt > num {
            r--
            continue
        }
        
        l++
    }
    
    return false
}